AutoGraph/WebSocketClient.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+import Starscream
+
+open class WebSocketClient {
+    public let baseUrl: String
+    public var httpHeaders: [String : String]
+    public var webSocket: WebSocket?
+    public var lifeCycle: GlobalLifeCycle?
+    private var webSocketListener: WebSocketListener?
+    
+    public init(baseUrl: String,
+                httpHeaders: [String : String] = [:]) {
+        self.baseUrl = baseUrl
+        self.httpHeaders = httpHeaders
+    }
+    
+    open func send<R: Request>(_ request: R, completion: @escaping RequestCompletion<R.SerializedObject>) {
+        do {
+            let query = try request.queryDocument.graphQLString()
+            var parameters: [String : Any] = ["query" : query]
+            if let variables = try request.variables?.graphQLVariablesDictionary() {
+                parameters["variables"] = variables
+            }
+            
+            guard let url = URL(string: self.baseUrl) else {
+                return
+            }
+            
+            let request = try URLRequest(url: url, method: .post)
+            
+            self.webSocket = WebSocket(request: request)
```

Comment:
what happens if we already have a websocket?

Reply:
Hmm ya good call did not account for that.